contractcourt/chain_watcher.go

```diff
@@ -30,20 +30,91 @@ const (
 	maxCommitPointPollTimeout = 10 * time.Minute
 )
 
-// LocalUnilateralCloseInfo encapsulates all the informnation we need to act
-// on a local force close that gets confirmed.
+// LocalUnilateralCloseInfo encapsulates all the informnation we need to act on
+// a local force close that gets confirmed.
 type LocalUnilateralCloseInfo struct {
 	*chainntnfs.SpendDetail
 	*lnwallet.LocalForceCloseSummary
 	*channeldb.ChannelCloseSummary
+
+	// CommitSet is the set of known valid commitments at the time the
+	// remote party's commitemnt hit the chain.
+	CommitSet CommitSet
 }
 
-// CooperativeCloseInfo encapsulates all the informnation we need to act
-// on a cooperative close that gets confirmed.
+// CooperativeCloseInfo encapsulates all the informnation we need to act on a
+// cooperative close that gets confirmed.
 type CooperativeCloseInfo struct {
 	*channeldb.ChannelCloseSummary
 }
 
+// RemoteUnilateralCloseInfo wraps the normal UnilateralCloseSummary to couple
+// the CommitSet at the time of channel closure.
+type RemoteUnilateralCloseInfo struct {
+	*lnwallet.UnilateralCloseSummary
+
+	// CommitSet is the set of known valid commitments at the time the
+	// remote party's commitment hit the chain.
+	CommitSet CommitSet
```

Comment:
Just the HTLCs atm, but I don't think there's any harm in passing the rest in. It's also nice since the channel arb now gains _full_ context on the valid commitment state at time of channel closure.

Reply:
In general, I think there is harm in passing in more info than needed. The extreme is making every variable global so all code can access everything.

My preference would be to pass in just what is needed and only change that when there is a real need for the channel arbitrator to know more.

Reply:
I don't think we're anywhere near that extreme. I don't consider this a blocking item, there're many other areas in the codebase where we use an entire struct for convenience when we only need a few fields. It's also the case that the chain watcher already passes in the entire commitment confirmed, we just modify it to also pass in all valid commitments at time of confirmation.